File: javastory.budgetsh.stage2/src/javastory/budgetsh/stage2/budget/ui/menu/CashBookMenu.java
package javastory.budgetsh.stage2.budget.ui.menu;

import java.util.Scanner;

import javastory.budgetsh.stage2.budget.entity.budget.CashBook;
import javastory.budgetsh.stage2.budget.ui.console.CashBookConsole;
import javastory.budgetsh.stage2.budget.util.Narrator;
import javastory.budgetsh.stage2.budget.util.TalkingAt;
import javastory.budgetsh.stage2.club.entity.club.TravelClub;

public class CashBookMenu {
	//
	private TransactionMenu transactionMenu;
	private CashBookConsole cashbookConsole;
	
	private Scanner scanner;
	private Narrator narrator;
	
	private CashBook currentBook;
	
	public CashBookMenu() {
		//
		scanner = new Scanner(System.in);
		narrator = new Narrator(this, TalkingAt.Left);
		
		transactionMenu = new TransactionMenu();
		cashbookConsole = new CashBookConsole();
	}
	
	public void show(TravelClub travelClub) {
		//
		int inputNumber = 0;
		
		while(true) {
			//
			displayMenu();
			inputNumber = selectMenu(); 
			
			switch (inputNumber) {
			//
			case 1:
				currentBook = cashbookConsole.register(travelClub);
				break;
			case 2:
				currentBook = cashbookConsole.find();
				break;
			case 3:
				currentBook = cashbookConsole.modify();
				break;
			case 4:
				cashbookConsole.remove();
				break;
			case 5:
				transactionMenu.show(currentBook);
				break;
			case 0:
				return;
				
			default:
				narrator.sayln("choose again!");
				continue;
			}
		}
	}
	
	public void displayMenu() {
		//
		narrator.sayln("------------------------------");
		if (currentBook!=null) {
			narrator.sayln(" CashBook Menu [ " + currentBook.getTravel().getName()+" ]");
		}else {
			narrator.sayln(" CashBook Menu [	No selected ]");
		}
		narrator.sayln("------------------------------");
		narrator.sayln(" 1. register");
		narrator.sayln(" 2. find");
		narrator.sayln(" 3. modify");
		narrator.sayln(" 4. remove");
		narrator.sayln("------------------------------");
		narrator.sayln(" 5. Transaction Menu");
		narrator.sayln("------------------------------");
		narrator.sayln(" 0. Previous Menu");
	}
	
	public int selectMenu() {
		//
		narrator.say("Select : ");
		int menuNumber = scanner.nextInt();
		
		if (menuNumber >= 0 && menuNumber <= 5) {
			scanner.nextLine();
			return menuNumber;
		} else {
			narrator.sayln(" It's invalid Number -->" + menuNumber);
		}
		return menuNumber;
	}
}

File: javastory.budgetsh.stage5/src/ui/menu/CashBookMenu.js
var co = require('co');
var prompt = require('co-prompt');
const TransactionMenu = require('./TransactionMenu.js');
const CashBookConsole = require('../console/CashBookConsole.js');
//const MainMenu = require('./MainMenu.js');
//
const cashbookConsole = new CashBookConsole();
const transactionMenu = new TransactionMenu();

class CashBookMenu {
	constructor() {
	}

	show(previousMenu) {
		this.selectMenu(previousMenu);
		//this.previousMenu = previousMenu;
	}

	displayMenu() {
		console.log("\n------------------------------");
		console.log("CashBook Menu");
		console.log("------------------------------");
		console.log(" 1. register");
		console.log(" 2. find");
		console.log(" 3. modify");
		console.log(" 4. remove");
		console.log("------------------------------");
		console.log(" 5. Transaction Menu");
		console.log("------------------------------");
		console.log(" 0. Previous Menu");
	}

	selectMenu(previousMenu) {
		var selectedNumber = '';

		this.displayMenu();
		co(function*() {
			var selectedNumber = yield prompt('>  ');
			var numberLine = Number(selectedNumber);
			if (numberLine > 0 && numberLine <= 5) {
				sendToService(numberLine);
			} else if (numberLine == 0) {
				previousMenu.show();
			} else {
				this.selectNumber(previousMenu);
			}
		});
	}

}
cashbookMenu = new CashBookMenu();

function sendToService(inputNumber) {
	var currentBook = null;
	switch (inputNumber) {
	//
	case 1:
		currentBook = cashbookConsole.register(cashbookMenu);
		break;
	case 2:
		currentBook = cashbookConsole.find(cashbookMenu);
		break;
	case 3:
		currentBook = cashbookConsole.modify(cashbookMenu);
		break;
	case 4:
		cashbookConsole.remove(cashbookMenu);
		break;
	case 5:
		transactionMenu.show(cashbookMenu);
		break;
	case 0:
		break;
	default:
		break;
	}
}

module.exports = CashBookMenu;